aten/src/ATen/cpu/vec/vec_base.h

```diff
@@ -739,8 +739,10 @@ template<class T, typename Op>
 static inline Vectorized<T> bitwise_binary_op(const Vectorized<T> &a, const Vectorized<T> &b, Op op) {
   static constexpr uint32_t element_no = VECTOR_WIDTH / sizeof(intmax_t);
   __at_align__ intmax_t buffer[element_no];
-  const intmax_t *a_ptr = reinterpret_cast<const intmax_t*>((const T*) a);
-  const intmax_t *b_ptr = reinterpret_cast<const intmax_t*>((const T*) b);
+  intmax_t a_ptr[element_no];
```

Comment:
A few thoughts:
1. if you use `std::array` instead of c-array, you can avoid the constexpr multiply in the memcpy below (you would do `sizeof(a_ptr)` instead).
2. this is no longer a pointer, so it should have a different name
3. I am concerned that the compiler won't be smart enough to avoid the copy when you copy the entire buffer like this. I suspect it will be more efficient if you `memcpy` a word at a time. Is this code performance sensitive? Can we run it through godbolt to see what it generates on different platforms?

Reply:
I updated this: `std::array` vs c-array isn't relevant to the `sizeof` change, but you ought to use `std::array` anyways.